### PR TITLE
Create heroic-tattoo.lic

### PR DESCRIPTION
**Pull request by Raykyn55**
Basic script you'd run at startup, calculates downtime for heroic tattoo based on your arcana, then attempts to keep it up at all times. Testing well for me.

### CODE REVIEW

**Comment by Raykyn55** [comment]
```--- Lich: afk paused.
[tattoo]>invoke my tattoo
You brace yourself as you activate your tattoo and feel empowered, as its magic washes over you.
>
Next attempt in 106 minutes
--- Lich: afk unpaused.
Gained: Arcana(+27)
```
chugging along nicely.

File: heroic-tattoo.lic [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+if args.cycle
+    DRC.message("Heroic Tattoo uptime is 60 minutes per activation")
+    DRC.message("Approximate downtime on your Heroic Tattoo is : #{downtime/60} minutes")
+    DRC.message("So total cycle from activation to next possible activation is #{(downtime/60)+60} minutes")
+    exit
+end
+
+pause 5
+loop do
+    while next_invoke > Time.now || no_use_scripts.any? { |name| Script.running?(name) } || ( no_use_rooms.any? { |room| room === DRRoom.title.to_s()[2..-3] || room == Room.current.id} )
```
Comment (asechrest) [inline heroic-tattoo.lic:28]: Can you help me out on what's going on in this complex while loop? Just trying to understand it.

File: heroic-tattoo.lic [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+no_use_rooms = get_settings.wand_watcher_no_use_rooms
+no_use_scripts = ['burgle','go2','athletics']
+downtime = DRSkill.getrank('Arcana') < 1400 ? (60 - DRSkill.getrank('Arcana')/24)*60 : 0
+timer = 3600 + downtime
+next_invoke = Time.now
+
+arg_definitions = [[
+      { name: 'once', regex: /once/i, optional: true, description: "One activation, however long that takes"},
+      { name: 'try', regex: /try/i, optional: true, description: "One ATTEMPT at activation, regardless of success or failure, and does not continue attempting"},
```
Comment (asechrest) [inline heroic-tattoo.lic:14]: What is the purpose of an invoke attempt that accepts a failure to invoke and doesn't care?

File: heroic-tattoo.lic [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    case DRC.bput("invoke my tattoo",
+        'as its magic washes over you',
+        'Its magic appears depleted',
+        'already under a heroic effect',
+        'Closing your eyes, you carefully bend')
+    when 'as its magic washes over you'
+        next_invoke = Time.now + timer
+        DRC.message("Next attempt in #{(next_invoke - Time.now).to_i/60} minutes")
+        break if args.once
```
Comment (asechrest) [inline heroic-tattoo.lic:41]: Are we properly unpausing the scripts that were paused when we `break` for args.once?

Comment (asechrest) [reply heroic-tattoo.lic:41]: Answering my own question: I think we do. It's a break not an exit. Thanks.

File: heroic-tattoo.lic (hunk 1 above)
Comment (Raykyn55) [reply heroic-tattoo.lic:28]: First check is timer, second is scripts running, third is room id/description. If any of those are true(timer not expired, script is running we dont want to interfere with, or we're in a room thats on the blacklist) we pause and check again in 15 seconds. All have to be false to pass to the loop.

File: heroic-tattoo.lic (hunk 2 above)
Comment (Raykyn55) [reply heroic-tattoo.lic:14]: T2 if folks dont want to run it all the time, but want to trigger it at the start of combat. T2 would wait for it to end, which could be a long time.

File: heroic-tattoo.lic (hunk 3 above)
Comment (Raykyn55) [reply heroic-tattoo.lic:41]: actually, this is a problem, it leaves the whole loop. works on try, not on once.

**Pull request merged**